resources/migrations/056_update_migrations.yaml

```diff
@@ -169,7 +169,320 @@ databaseChangeLog:
+              - column:
+                  remarks: The timestamp of when the app was created
+                  name: created_at
+                  type: ${timestamp_type}
+                  defaultValueComputed: current_timestamp
+                  constraints:
+                    nullable: false
+              - column:
+                  remarks: The timestamp of when the app was updated
+                  name: updated_at
+                  type: ${timestamp_type}
+                  defaultValueComputed: current_timestamp
+                  constraints:
+                    nullable: false
+      rollback:
+        - dropTable:
+            tableName: data_app
+
+  - changeSet:
+      id: v56.2025-07-03T16:01:00
+      author: qnkhuat
+      comment: Create data_app_definition table for data apps functionality
+      preConditions:
+        - onFail: MARK_RAN
+        - not:
+            - tableExists:
+                tableName: data_app_definition
+      changes:
+        - createTable:
+            tableName: data_app_definition
+            remarks: App definitions (append only)
+            columns:
+              - column:
+                  name: id
+                  remarks: Unique ID
+                  type: int
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: app_id
+                  remarks: Referenced app
+                  type: int
+                  constraints:
+                    nullable: false
+                    referencedTableName: data_app
+                    referencedColumnNames: id
+                    foreignKeyName: fk_data_app_definition_app_id
+                    deleteCascade: true
+              - column:
+                  name: revision_number
+                  remarks: Revision number for ordering
+                  type: int
+                  constraints:
+                    nullable: false
+              - column:
+                  name: config
+                  remarks: App configuration JSON
+                  type: ${text.type}
+                  constraints:
+                    nullable: false
+              - column:
+                  name: entity_id
+                  remarks: SHA hash for the entity
+                  type: char(21)
+                  constraints:
+                    nullable: true
+              - column:
+                  remarks: The timestamp of when the definition was created
+                  name: created_at
+                  type: ${timestamp_type}
+                  defaultValueComputed: current_timestamp
+                  constraints:
+                    nullable: false
+      rollback:
+        - dropTable:
+            tableName: data_app_definition
+
+  - changeSet:
+      id: v56.2025-07-03T16:02:00
+      author: qnkhuat
+      comment: Add index on data_app_definition.app_id
```

Comment:
I don't think we need this index, since the database engines should give it for free as its a prefix of the larger index.

At least in MySQL and Postgres this is how it works, and I don't think we need H2 to scale.

Reply:
it doesn't iirc, I have to add a test to remind people writing an create index for fk a while back: https://github.com/metabase/metabase/pull/33578